tests/slurm_plugin/test_clustermgtd.py

```diff
@@ -2112,6 +2170,794 @@ def test_update_status(self, compute_fleet_status_manager, put_item_response, ex
+                SlurmNode("queue1-dy-c5xlarge-3", "ip-3", "hostname", "some_state", "queue1"),
+                SlurmNode("queue1-st-c5xlarge-999", "ip-1", "hostname", "some_state", "queue1"),
+            ],
+            set(),
+            {NodeType("queue1-c5xlarge", "queue1")},
+        ),
+        (
+            [SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1")],
+            {NodeType("queue1-c5xlarge", "queue1")},
+            {NodeType("queue1-c5xlarge", "queue1")},
+        ),
+        (
+            [
+                SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1"),
+                SlurmNode("queue2-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue2"),
+            ],
+            {NodeType("queue1-c5xlarge", "queue1")},
+            {NodeType("queue1-c5xlarge", "queue1"), NodeType("queue2-c5xlarge", "queue2")},
+        ),
+    ],
+)
+@pytest.mark.usefixtures("initialize_instance_manager_mock", "initialize_compute_fleet_status_manager_mock")
+def test_handle_bootstrap_failure_nodes(
+    bootstrap_failure_nodes, expected_bootstrap_failure_nodes_types, initial_bootstrap_failure_nodes_types, mocker
+):
+    cluster_manager = ClusterManager(mocker.MagicMock())
+    mock_increase_partitions_protected_failure_count = mocker.patch(
+        "slurm_plugin.clustermgtd.ClusterManager._increase_partitions_protected_failure_count"
+    )
+    cluster_manager._bootstrap_failure_nodes_types = initial_bootstrap_failure_nodes_types
+    cluster_manager._handle_bootstrap_failure_nodes(bootstrap_failure_nodes)
+    assert_that(mock_increase_partitions_protected_failure_count.call_count).is_equal_to(1)
+    assert_that(cluster_manager._bootstrap_failure_nodes_types).is_equal_to(expected_bootstrap_failure_nodes_types)
+
+
+@pytest.mark.parametrize(
+    "node, private_ip_to_instance_map, is_backing_instance_valid, current_node_in_replacement, "
+    "bootstrap_failure_messages, is_node_bootstrap_failure",
+    [
+        (
+            SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            False,
+            {"queue1-st-c5xlarge-1"},
+            "Node bootstrap error: Node is currently in replacement and no backing instance",
+            True,
+        ),
+        (
+            SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", datetime(2020, 1, 1, 0, 0, 0)),
+            },
+            True,
+            {"queue1-st-c5xlarge-1"},
+            "Node bootstrap error: Replacement timeout expires for node in replacement",
+            True,
+        ),
+        (
+            SlurmNode("queue1-dy-c5xlarge-1", "ip-1", "hostname", "#MIXED", "queue1"),
```

Comment:
Let's try to use real slurm states. Same comment applies to all test cases in this test